server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetAliasesAction.java

```diff
@@ -89,56 +83,59 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         getAliasesRequest.indicesOptions(IndicesOptions.fromRequest(request, getAliasesRequest.indicesOptions()));
         getAliasesRequest.local(request.paramAsBoolean("local", getAliasesRequest.local()));
 
-        //we may want to move this logic to TransportGetAliasesAction but it is based on the original provided aliases, which will
-        //not always be available there (they may get replaced so retrieving request.aliases is not quite the same).
```

Comment:
This ties in with the other comments.

I think the RestHandler here tries to identify which of the many URl patterns have been invoked. For example, if the URI had index names, make sure those index names are printed even if they have no aliases. On the other hand, if aliases were requested (maybe alongside indices), than make sure all the requested aliases are returned and no empty indices without aliases will be returned.

I don't agree that the TransportAction should handle this types of corner cases. I think the TransportAction should return a minimal set of aliases (no extraneous placeholders) and the REST handler should package it with empty or null placeholders as well as to do any existence checks that are required to meet each individual REST spec.

> TransportGetAliasesAction but it is based on the original provided aliases, which will not always be available there (they may get replaced so retrieving request.aliases is not quite the same).

`GetAliasesRequest#originalAliases` contains the original provided aliases, but again I don't think **a single transport action can handle several different rest actions, especially when those rest actions have conflicting requirements, like for example the inclusion of indices with no aliases**.

Reply:
The other comment above is more about "we have duplicated logic which should be in one place, but there's a good reason". This comment is more about whether some parts of the REST action should be moved to the transport action. I generally prefer rest actions with almost no logic, but that has to do with trying to keep rest responses and transport client responses the same and have REST layer only doing the printing. With the transport client being deprecated, this is less and less important. 

The reason why this comment is there is because we have tried in the past to move all the logic to the transport action, but that did not quite work, also it made bw compatibility hard. We have added this comment to remember especially that `request.aliases` may get replaced, which is a tricky thing that can be easily missed. `originalAliases` were not there yet when the comment was written which means that it's a good time now to re-evaluate if this comment is still valid. 

In general, this API does quite unique things, I always found it weird that the REST layer had more information about the request than transport (till originalAliases were added to the request) and I had envisioned that one day maybe it would become more similar to the other API, with logic in the transport action and a very simple REST action. But I see how this comment does not add much value at this point, so I am fine with it being removed. Maybe we should re-evaluate what changes we want to make long-term to this API in a different PR though. See also #30536 .